### PR TITLE
Chameleon ID and PDA improvements

### PR DESCRIPTION
**Pull request by Alek2ander**
## About The Pull Request
Chameleon IDs and PDAs will automatically update names and icons when changing disguise
They will also change the displayed job when switching outfits with the chameleon combo-action

## Why It's Good For The Game
Chameleon IDs and PDAs reset to the default name if their appearance changes and need to be updated manually to display the proper label, which seems like a waste of time to me.
Changing the entire chameleon loadout is very sneaky and setting the appropriate job on the ID and PDA makes it sneakier.

## Changelog
:cl:
tweak: Agent IDs and chameleon PDAs now automatically update their labels and icons when transforming.
tweak: Selecting a chameleon outfit will set equipped agent IDs and chameleon PDAs to display the appropriate job.
/:cl:

### CODE REVIEW

File: code/modules/clothing/chameleon.dm [hunk 1]
```diff
@@ -247,6 +250,39 @@
 		return
 	random_look(owner)
 
+/datum/action/item_action/chameleon/change/proc/apply_job_data(datum/job/job_datum)
+	return
+
+/datum/action/item_action/chameleon/change/id
```
Comment (actioninja) [inline code/modules/clothing/chameleon.dm]: drop the explicit type definition, byond will create it implicitly when you override the procs below

File: code/modules/clothing/chameleon.dm [hunk 2]
```diff
@@ -247,6 +250,39 @@
...
+	if(istype(agent_card))
+		agent_card.update_label()
+
+/datum/action/item_action/chameleon/change/id/apply_job_data(datum/job/job_datum)
+	..()
+	var/obj/item/card/id/agent_card = target
+	if(istype(agent_card))
+		agent_card.assignment = job_datum.title
+
+/datum/action/item_action/chameleon/change/pda
```
Comment (actioninja) [inline code/modules/clothing/chameleon.dm]: same

File: code/modules/clothing/chameleon.dm (hunk 1 above)
Comment (actioninja) [inline code/modules/clothing/chameleon.dm]: drop the explicit type definition, byond will create it implicitly when you override the procs below
forgot to make these a requested change

File: code/modules/clothing/chameleon.dm (hunk 2 above)
Comment (actioninja) [inline code/modules/clothing/chameleon.dm]: same

**Review by actioninja — CHANGES_REQUESTED** [review]

File: code/modules/clothing/chameleon.dm (hunk 1 above)
Comment (Alek2ander) [reply code/modules/clothing/chameleon.dm]: 👍

File: code/modules/clothing/chameleon.dm (hunk 2 above)
Comment (Alek2ander) [reply code/modules/clothing/chameleon.dm]: 👍

**Review by actioninja — APPROVED** [review]

**Pull request merged**